SU-SWS/saa_victoria_subtheme
# Adapt 1195  icon cards variant

# ALMOST READY FOR REVIEW

# Summary
-  Define 2 basic card variant fields and begin adjusting cards to support icon and other types.

# Review By (Date)
- 2/23/21

# Urgency
- How critical is this PR? Low

# Steps to Test

1. Pull down the branch or test on https://saaalumnicenterevents-test.sites.stanford.edu/ 
2. Clear the site cache
3. Check that existing cards (that haven't been edited by you) are receiving icon styling as expected - see homepage of test site above.

![image](https://user-images.githubusercontent.com/12898896/108755797-ece9db00-7504-11eb-8ad1-e04569f74b4f.png)

4. Edit that card on the homepage, and WITHOUT SAVING, check that the default values are showing on the styles tab.
5. Make sure that when editing you can see 2 fields (card type and card border style) as expected.
6. On a new unpublished test page, create and edit some cards. Try out all field options and make sure that the correct field values are being added to the element 2 levels above the element with class 'su-card'. 
7. Test: Right now you can visually see when a card is set to a non-icon type, as the icon colored tab in the corner will not display.
8. Note that while the border or minimal value SHOULD apply correctly to the parent element, it will always have a dropshadow edge right now as the styles for that field aren't implemented until a later ticket.

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?
- We will not be pushing this until a later time, but if it was pushed, this PR should not affect live cards. We have tested that above ^

# Associated Issues and/or People
- JIRA ticket ADAPT-1195 ADAPT-1836 ADAPT-1520
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
